File: backend/database/cassandra_db.js
require("dotenv").config();
const { Client } = require("cassandra-driver");
const path = require("path");

const credentials = {
  clientId: process.env.CASSANDRA_CLIENT_ID,
  secret: process.env.CASSANDRA_SECRET,
  token: process.env.CASSANDRA_TOKEN,
};

async function runCassandraDB() {
  const client = new Client({
    cloud: {
      secureConnectBundle: path.resolve(
        __dirname,
        "secure-connect-playlist-db.zip"
      ),
    },
    credentials: {
      username: credentials.clientId,
      password: credentials.secret,
    },
  });

  await client.connect();

  // Execute a query
  const rs = await client.execute("SELECT * FROM playlist.playlist_tb");
  console.log(`Your cluster returned ${rs.rowLength} row(s)`);
  console.dir(rs);

  let properties = rs.columns.map((element) => element.name);
  let tabularData = rs.rows;
  console.log(tabularData, properties);
  console.table(tabularData);

  await client.shutdown();
}

// Run the async function
module.exports = runCassandraDB;

File: pyend/database/cassandraDB.py
import asyncio
from cassandra.io.libevreactor import LibevConnection
from cassandra.cluster import Cluster
from cassandra.auth import PlainTextAuthProvider
import json


async def main():
    cloud_config = {"secure_connect_bundle": "secure-connect-playlist-db.zip"}

    with open("playlist_db-token.json") as f:
        secrets = json.load(f)

    CLIENT_ID = secrets["clientId"]
    CLIENT_SECRET = secrets["secret"]

    auth_provider = PlainTextAuthProvider(CLIENT_ID, CLIENT_SECRET)
    cluster = await Cluster.connect_async(
        cloud=cloud_config, auth_provider=auth_provider
    )
    cluster.connection_class = LibevConnection()
    session = cluster.connect()

    row = await session.execute_async("SELECT * FROM playlist.playlist_tb").one()
    if row:
        print(row[0])
    else:
        print("An error occurred.")
